### ISSUE TITLE
Server errors result in the _error page displaying a 404 and the original err is missing.

### ISSUE THREAD

**Issue by HackedByChinese** [kyle-mccarthy/nest-next#7]
I've noticed this line in `render.filter`: https://github.com/kyle-mccarthy/nest-next/blame/master/lib/render.filter.ts#L28. If err has a property `response`, it tries to render the request normally rather than using the error renderer. Is there a reason for this?

I've noticed it's pretty common for server errors to have this property set, and it causes these server errors to show up as seemingly inexplicable 404s with the original error lost. For example, if I'm using a simple global guard to ensure the current request is authenticated, and the request is _not_ authenticated, Nest throws a custom exception that results in a 403, and this error object has `response` populated, and this causes the described problem.

Example guard:

```
import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common'
import { Request } from 'express'
import { Observable } from 'rxjs'

@Injectable()
export class AuthenticatedGuard implements CanActivate {
  canActivate(
    context: ExecutionContext
  ): boolean | Promise<boolean> | Observable<boolean> {
    const req = context.switchToHttp().getRequest<Request>()

    return req.isAuthenticated() // this is added by passport. if you don't have passport, you can synthesize the error conditions with return false instead.
  }
}
```

In your server's `main.ts`, simply add `server.useGlobalGuards(new AuthenticatedGuard())`. Visiting any URL should manifest a 404.

Other examples: When an axios request generates an error, the err object also has a `response` property.

In some cases you can work around this issue by clearing the response property, assuming you can intercept the error before it passes on to the render.filter, but that's not always possible.

**Comment by kyle-mccarthy** [comment]
Hey good question, the reason that its done this way it to allow for the next assets (ex `/_next/static/development/pages/_app.js`) to be rendered by its own renderer. However, you are right that it is swallowing errors that shouldn't be handled by next. 

Let me look into this a little more, but my initial thought is to allow errors with the 404 status code to be passed to the normal render function and direct all other errors to the next's error handler.

**Comment by HackedByChinese** [comment]
Ah, I see. Thanks for the feedback (and also for your work on this module, it's great).

**Comment by kyle-mccarthy** [comment]
Thanks, I am glad that you have been liking it so far!

I just pushed v0.5 which contains a much different process to handling errors and also allow for you to hook into the RenderFilter and modify the behavior/prevent next for responding. 

By default next  just forwards the error to the _error page, but that doesn't really take into account if you want to return JSON or something else instead. To address that, I added an optional errorHandler to the RenderService, which is called before sending the error the next.

The relevant docs are [here](https://github.com/kyle-mccarthy/nest-next#handling-errors] and changes are in [this commit](https://github.com/kyle-mccarthy/nest-next/commit/e48d80859ee49323eebe4575ef61b5fbc3d15f3c). Let me know if that helps address your original problems and/or you have any other issues!
